### PR TITLE
Enable nullability in Glyph classes and BehaviorService

### PR DESCRIPTION
**Pull request by halgab**
## Proposed changes

- Enable nullability in `Glyph`, `GlyphCollection`, `ToolStripItemGlyph` and `BehaviorService`

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/9341)

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs [hunk 1]
```diff
@@ -82,12 +80,13 @@ internal BehaviorService(IServiceProvider serviceProvider, Control windowFrame)
...
 
         // Create out object that will handle all menucommands
-        if (serviceProvider.GetService(typeof(IMenuCommandService)) is IMenuCommandService menuCommandService
-            && serviceProvider.GetService(typeof(IDesignerHost)) is IDesignerHost host)
+        IMenuCommandService? menuCommandService = serviceProvider.GetService<IMenuCommandService>();
+        IDesignerHost? host = serviceProvider.GetService<IDesignerHost>();
+        if (menuCommandService is not null && host is not null)
         {
-            _menuCommandHandler = new MenuCommandHandler(this, menuCommandService);
+            MenuCommandHandler menuCommandHandler = new MenuCommandHandler(this, menuCommandService);
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs:87]:
```suggestion
            MenuCommandHandler menuCommandHandler = new(this, menuCommandService);
```

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs [hunk 2]
```diff
@@ -469,14 +458,11 @@ public void PushCaptureBehavior(Behavior behavior)
         // VSWhidbey #373836. Since we are now capturing all mouse messages, we might miss some WM_MOUSEACTIVATE
         // which would have activated the app. So if the DialogOwnerWindow (e.g. VS) is not the active window,
         // let's activate it here.
-        IUIService uiService = (IUIService)_serviceProvider.GetService(typeof(IUIService));
-        if (uiService is not null)
+        IUIService? uiService = _serviceProvider.GetService<IUIService>();
+        IWin32Window? hwnd = uiService?.GetDialogOwnerWindow();
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs:462]: hwnd - > hWnd

Comment (dreddy-work) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs:462]: technically, existing code is more optimized.  `hwnd` won't be declared if uiService is null.

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs [hunk 3]
```diff
@@ -589,25 +572,23 @@ private MenuCommand FindCommand(CommandID commandID, IMenuCommandService menuSer
...
     }
 
     private void ShowError(Exception ex)
     {
-        if (_serviceProvider.GetService(typeof(IUIService)) is IUIService uis)
-        {
-            uis.ShowError(ex);
-        }
+        IUIService? uis = _serviceProvider.GetService<IUIService>();
+        uis?.ShowError(ex);
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs:583]:
```suggestion
        _serviceProvider.GetService<IUIService>()?.ShowError(ex);
```

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs [hunk 4]
```diff
@@ -205,7 +194,7 @@ internal DragDropEffects DoDragDrop(DropSourceBehavior dropSourceBehavior)
         DropSource.QueryContinueDrag += new QueryContinueDragEventHandler(dropSourceBehavior.QueryContinueDrag);
         DropSource.GiveFeedback += new GiveFeedbackEventHandler(dropSourceBehavior.GiveFeedback);
 
-        DragDropEffects res = DragDropEffects.None;
+        DragDropEffects res;
```
[the remark this answers is not in the document]
Comment (halgab) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs:197]: No, I didn't have any warnings. I just think that not setting an unused value marks intention better, but it's a matter of personal preference. I'll rollback

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs [hunk 5]
```diff
@@ -557,15 +540,15 @@ private bool PropagateHitTest(Point pt)
...
     internal void StartDragNotification() => _adornerWindow.StartDragNotification();
 
-    private MenuCommand FindCommand(CommandID commandID, IMenuCommandService menuService)
+    private MenuCommand? FindCommand(CommandID commandID, IMenuCommandService menuService)
     {
-        Behavior behavior = GetAppropriateBehavior(_hitTestedGlyph);
+        Behavior? behavior = GetAppropriateBehavior(_hitTestedGlyph);
 
         if (behavior is not null)
         {
```
[the remark this answers is not in the document]
Comment (halgab) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs:548]: Reading it again, nothing obvious comes to mind unfortunately

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs (hunk 2 above)
Comment (halgab) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs:462]: I tried to keep it short and optimized with pattern matching to address your comment

**Pull request merged**